src/prompts/index.tsx

```diff
@@ -55,12 +54,20 @@ export class PromptManager {
 
         this.pushContainer = rootFrame.createContainer('push');
         this.context = ctx;
+        this.pushManager = pushManager;
 
         this.setState('loading');
     }
 
-    private onEventTracked = (e: SdkEvent) => {
-        console.info('Prompt trigger saw event', e);
+    private onEventTracked = (event: KumulosEvent) => {
+        console.info('Prompt trigger saw event', event);
+
+        if (event.type === EventType.PUSH_UNREGISTERED) {
+            console.info('Prompt identified an unregistered event');
+            this.activePrompts = [];
+            this.evaluateTriggers();
```

Comment:
Is there a reason we re-evaluate triggers, could it result in some prompts becoming active again? (I guess likely not, there won't be any trigger events queued to look at?)

Reply:
The goal was to unify the flow, no reason to re evaluate. 
Extracted to a method that just fetches the current permission state and then re-renders